raven/base.py

```diff
@@ -636,7 +636,13 @@ def capture(self, event_type, data=None, date=None, time_spent=None,
             **kwargs)
 
         # should this event be sampled?
-        if self._random.random() < self.sample_rate:
+        sample_rate = self.sample_rate
+        try:
+            sample_rate = float(extra['sample_rate'])
```

Comment:
> I would probably do a get here instead of exception

Do you mean something like: 
`sample_rate = float((extra or {}).get('sample_rate', self.sample_rate))`?

> we'd want this to be an arg to capture rather than part of the extra metadata

hmm, I was thinking the usage would be something like: 
`some_logger.error("asd", extra={'sample_rate': 0.4})`

I'll need a bit of time to dig into the codebase to figure out how to make that work without putting the sample rate in the `extra`